payment-initiation-api/src/main/java/com/aevi/sdk/pos/flow/model/BasketItemModifierBuilder.java

```diff
@@ -0,0 +1,91 @@
+package com.aevi.sdk.pos.flow.model;
+
+import android.support.annotation.NonNull;
+
+import static com.aevi.sdk.flow.util.Preconditions.checkArgument;
+import static com.aevi.sdk.flow.util.Preconditions.checkNotEmpty;
+
+/**
+ * A builder to create {@link BasketItemModifier} instances
+ */
+public class BasketItemModifierBuilder {
+
+    private final String name;
+    private final String type;
+
+    private String id;
+    private Float amount;
+    private Float percentage;
+
+    public BasketItemModifierBuilder(@NonNull String name, @NonNull String type) {
+        this.name = name;
+        this.type = type;
+    }
+
+    /**
+     * Set the id for this item modifier (optional).
+     *
+     * @param id The id
+     * @return This builder
+     */
+    @NonNull
+    public BasketItemModifierBuilder withId(String id) {
+        this.id = id;
+        return this;
+    }
+
+    /**
+     * Set the absolute amount of this modifier in subunit form (e.g pence / cents). This amount may be fractional i.e. fractions of one cent/penny.
+     * A fractional value may used in cases such as representing a tax amount that needs to be correct to several decimal places of accuracy when
+     * these item modifiers are added together as a part of a whole transaction.
```

Comment:
Maybe have a reference here to the other method as well - "If fractional values are not required, see withAmount" or whatevs?